hca/util/__init__.py

```diff
@@ -131,16 +142,15 @@ def _request(self, req_args, url=None, stream=False, headers=None):
         query = {k: v for k, v in req_args.items()
                  if self.parameters.get(k, {}).get("in") == "query" and v is not None}
         body = {k: v for k, v in req_args.items() if k in self.body_props and v is not None}
-        if "security" in self.method_data:
-            session = self.client.get_authenticated_session()
-        else:
-            session = self.client.get_session()
+        res = self._make_authenticated_request(body, headers, url, query, stream)
+
+        keep_session_alive = True
+        while keep_session_alive:
+            res = self._make_authenticated_request(body, headers, url, query, stream)
+            # DSSException 401 is returned and the jwt token expiration is only preserved in the stack trace
+            if not ('jwt.exceptions.ExpiredSignatureError: Signature has expired' in res.json().get('stacktrace', '')):
```

Comment:
Should we be relying on scraping data from a stack trace? This doesn't feel reliable.

Reply:
I don't like it either, even though I think the edge cases where this is unreliable are unlikely.  I will try to find a better way, but the request itself doesn't currently seem to have any other way to fetch it so I will probably have to rework how the errors are saved and propagated back.

Reply:
you should inspect the JWT locally to see when and if it's expired.